test/groovy/TestsPublishResultsTest.groovy

```diff
@@ -0,0 +1,112 @@
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertTrue
+
+import util.Rules
+
+class TestsPublishResultsTest extends BasePipelineTest {
+    Map publisherStepOptions
+    List archiveStepPatterns
+
+    @Rule
+    public RuleChain ruleChain = RuleChain.outerRule(Rules.getCommonRules(this))
+
+    def stepUnderTest
+
+    @Before
+    void init() {
+        publisherStepOptions = [:]
+        archiveStepPatterns = []
+        // prepare checkResultsPublish step
+        stepUnderTest = loadScript('testsPublishResults.groovy').testsPublishResults
+        // add handler for generic step call
+        helper.registerAllowedMethod('step', [Map.class], {
+            parameters -> publisherStepOptions[parameters.$class] = parameters
+        })
+        helper.registerAllowedMethod('junit', [Map.class], {
+            parameters -> publisherStepOptions['junit'] = parameters
+        })
+        helper.registerAllowedMethod('jacoco', [Map.class], {
+            parameters -> publisherStepOptions['jacoco'] = parameters
+        })
+        helper.registerAllowedMethod('cobertura', [Map.class], {
+            parameters -> publisherStepOptions['cobertura'] = parameters
+        })
+        helper.registerAllowedMethod('archiveArtifacts', [Map.class], {
```

Comment:
we may create a Rule for `archiveArtifacts` further down the road since we use this at several places.